### PR TITLE
Added Progressbar Base Component

### LINKED ISSUE #286: Progressbar
see #260

### PR DESCRIPTION
**Pull request by ClientCrash**
This pr uses divs because styling of the `progress` element isnt supported in all browsers. At least to the extend needed for this component.

- [x] Display Progressbar 
- [x] Progressbar should display the correct value. 
Closes #286

### CODE REVIEW

File: src/app/design/progressbar/progressbar.component.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<div class="bar">
+  <p>{{value}}%</p>
+  <p class="desc"><ng-content></ng-content></p>
+<div class="styled-p-bar-v2">
```
Comment (Tristan-H11) [inline src/app/design/progressbar/progressbar.component.html:4]:
```suggestion
<div class="styled-p-bar">
```

File: src/app/design/progressbar/progressbar.component.html [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  <div class="styled-p-bar-v2-value" style="width:{{value}}%">&nbsp;</div>
```
Comment (Tristan-H11) [inline src/app/design/progressbar/progressbar.component.html:5]:
```suggestion
  <div class="styled-p-bar-value" style="width:{{value}}%">&nbsp;</div>
```

File: src/app/design/progressbar/progressbar.component.scss [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  padding: 8px;
+}
+.desc{
+  font-size: 1rem;
+  color: grey;
+  margin-top: 0px;
+  margin-bottom: 4px;
+}
+
+.styled-p-bar-v2{
```
Comment (Tristan-H11) [inline src/app/design/progressbar/progressbar.component.scss:22]:
```suggestion
.styled-p-bar{
```

File: src/app/design/progressbar/progressbar.component.scss [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  height: 1.8vh;
+  width: 100%;
+  border-radius: 100vw;
+  background-color: white;
+  margin-top: 5px;
+  overflow: hidden;
+}
+.styled-p-bar-v2-value{
```
Comment (Tristan-H11) [inline src/app/design/progressbar/progressbar.component.scss:30]:
```suggestion
.styled-p-bar-value{
```

**Review by Tristan-H11 — CHANGES_REQUESTED** [review]
Is this v2-prefix really necessary?

File: src/app/design/progressbar/progressbar.component.html [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<div class="bar">
+  <p>{{value}}%</p>
+  <p class="desc"><ng-content></ng-content></p>
+<div class="styled-p-bar">
+  <div class="styled-p-bar-value" style="width:{{value}}%">&nbsp;</div>
```
Comment (MarcelCoding) [inline src/app/design/progressbar/progressbar.component.html:5]:
```suggestion
  <div class="styled-p-bar-value" [style.width.%]="width">&nbsp;</div>
```

File: src/app/app.module.ts [hunk 6]
```diff
@@ -111,7 +112,8 @@ const routes: Routes = [
     WalletAppTransactionComponent,
     HardwareShopSidebarComponent,
     EditorComponent,
-    FileManagerComponent
+    FileManagerComponent,
+    ProgressbarComponent
```
Comment (MarcelCoding) [inline src/app/app.module.ts:116]: Please define the progress bar component inside of the design module.

**Review by MarcelCoding — CHANGES_REQUESTED** [review]

File: src/app/design/progressbar/progressbar.component.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import { Component, Input} from '@angular/core';
+
+@Component({
+  selector: 'app-progressbar',
```
Comment (Tristan-H11) [inline src/app/design/progressbar/progressbar.component.ts:4]:
```suggestion
  selector: 'design-progressbar',
```

Comment (Tristan-H11) [reply src/app/design/progressbar/progressbar.component.ts:4]: We decided to give base-components the prefix "design".

**Review by Tristan-H11 — APPROVED** [review]

File: src/app/design/progressbar/progressbar.component.ts (hunk 7 above)
Comment (MarcelCoding) [reply src/app/design/progressbar/progressbar.component.ts:4]: Please don't, this only caused problems...

Comment (Tristan-H11) [reply src/app/design/progressbar/progressbar.component.ts:4]: Convinced Marcel. Design Prefix is fine, as long as there is an es-lint ignore annotation.

Comment (MarcelCoding) [reply src/app/design/progressbar/progressbar.component.ts:4]: That's bad, either you can tweak the eslint config or we don't do it.

Comment (Tristan-H11) [reply src/app/design/progressbar/progressbar.component.ts:4]: See https://github.com/cryptic-game/frontend/pull/346/commits/2a71c442c88f0ad45acdc2740ec3d20a28d84cc5.

**Pull request closed without merging**